pymbar/mbar.py

```diff
@@ -291,6 +291,55 @@ def getWeights(self):
         return np.exp(self.Log_W_nk)
 
     #=========================================================================
+    def computeOverlap(self, output='scalar'):
+        """Compute estimate of overlap matrix between the states.
+
+        Returns
+        -------
+        O : np.ndarray, float, shape=(K, K)
+            estimated state overlap matrix: O[i,j] is an estimate
+            of the probability of observing a sample from state i in state j
+
+        Parameters
+        ----------
+        output : string, optional
+            One of 'scalar', 'matrix', 'eigenvalues', 'all', specifying
+        what measure of overlap to return
+
+        Notes
+        -----
+
+        W.T * W \approx \int (p_i p_j /\sum_k N_k p_k)^2 \sum_k N_k p_k dq^N
+                      = \int (p_i p_j /\sum_k N_k p_k) dq^N
+
+        Multiplying elementwise by N_i, the elements of row i give the probability
+        for a sample from state i being observed in state j.
+
+        Examples
+        --------
+
+        >>> from pymbar import testsystems
+        >>> [x_kn, u_kn, N_k] = testsystems.HarmonicOscillatorsTestCase().sample(mode='u_kn')
```

Comment:
This line should have an extra variable `s_n` that represents the state of origin.  Our testsystems now return that as well.  E.g.:

```
(x_n, u_kn, N_k, s_n) = testsystems.HarmonicOscillatorsTestCase().sample(mode='u_kn')
```